code/datums/mind.dm

```diff
@@ -747,6 +747,8 @@
 		if(!new_objective)
 			return
 
+		new_objective.post_admin_creation()
```

Comment:
Actually, I don't think this is needed. I would rather you just typecast the escape objective and then assign the real target name:
```dm
new_objective = new /datum/objective/escape/escape_with_identity
new_objective.owner = src
new_objective.target = new_target
var/datum/objective/escape/escape_with_identity/O = new_objective
target_real_name = target.current.real_name
```

Reply:
For the love of god, why does this work? DM is a hellhole. Implemented in most recent commit.